File: controllers/userController.py
from fastapi import APIRouter, Depends
from models import User
from services.userService import get_user

router = APIRouter()

@router.get('/users/{user_id}', response_model=User)
async def read_user(user_id: int):
    user = get_user(user_id)
    if user is None:
        raise HTTPException(status_code=404, detail='User not found')
    return user

File: controllers/userController.js
const userService = require('../services/userService');

exports.getUser = async (req, res) => {
	try {
		const user = await userService.getUser(req.params.id);
		res.status(200).json(user);
	} catch (error) {
		res.status(404).json({ message: error.message });
	}
};